cli/sub_command.go

```diff
@@ -77,12 +81,67 @@ func configureSubCommand(app commandHost) {
 	act.Flag("stream", "Subscribe to a specific stream (required JetStream)").PlaceHolder("STREAM").StringVar(&c.stream)
 	act.Flag("ignore-subject", "Subjects for which corresponding messages will be ignored and therefore not shown in the output").Short('I').PlaceHolder("SUBJECT").StringsVar(&c.ignoreSubjects)
 	act.Flag("wait", "Max time to wait before unsubscribing.").DurationVar(&c.wait)
+	act.Flag("report-subjects", "Subscribes to a subject pattern and builds a de-duplicated report of active subjects receiving data").UnNegatableBoolVar(&c.reportSubjects)
+	act.Flag("report-top", "Number of subjects to show when doing 'report-subjects'. Default is 10.").Default("10").IntVar(&c.reportSubjectsCount)
 }
 
 func init() {
 	registerCommand("sub", 17, configureSubCommand)
 }
 
+func startSubjectReporting(ctx context.Context, subjMu *sync.Mutex, subjectReportMap map[string]int64, subjectBytesReportMap map[string]int64, subjCount int) {
+	go func() {
+		ticker := time.NewTicker(time.Second)
+
+		for range ticker.C {
```

Comment:
better to do like

```
for {
  select {
    case <-ticker.C:
    case ctx.Done():
  }
}
```

Reply:
Done: https://github.com/nats-io/natscli/pull/738/commits/e49fd63a0929654078661894fad1d5313a8be3a4